components/brave_wallet/common/f4_jumble.cc

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/common/f4_jumble.h"
+
+#include <algorithm>
+#include <array>
+#include <memory>
+#include <string>
+
+#include "base/notreached.h"
+#include "brave/components/brave_wallet/common/hex_utils.h"
+#include "brave/third_party/argon2/src/src/blake2/blake2.h"
+
+namespace brave_wallet {
+
+const size_t kLeftSize = 64u;
+
+std::string GetHPersonalizer(uint8_t i) {
+  return {85,  65, 95,  70,  52, 74, 117,
+          109, 98, 108, 101, 95, 72, static_cast<char>(i),
+          0,   0};
+}
+
+std::string GetGPersonalizer(uint8_t i, uint16_t j) {
+  return {85,
+          65,
+          95,
+          70,
+          52,
+          74,
+          117,
+          109,
+          98,
+          108,
+          101,
+          95,
+          71,
+          static_cast<char>(i),
+          static_cast<char>(j & 0xFF),
+          static_cast<char>(j >> 8)};
+}
+
+std::optional<std::vector<uint8_t>> blake2b(const std::vector<uint8_t>& payload,
+                                            std::string_view personalizer,
+                                            size_t len) {
+  blake2b_state blake_state = {};
+  blake2b_param params = {};
+
+  if (personalizer.length() != sizeof(params.personal)) {
+    NOTREACHED();
+    return std::nullopt;
+  }
+
```

Comment:
Personalizer should be a fixed size span, and this check should become a CHECK

Reply:
Done